src/solvers/strings/string_constraint_generator_main.cpp

```diff
@@ -114,6 +114,19 @@ void merge(string_constraintst &result, string_constraintst other)
     std::back_inserter(result.not_contains));
 }
 
+and_exprt char_set_restrictions(const exprt &expr, const std::string &char_set)
+{
+  // Parse char_set
+  PRECONDITION(char_set.length() == 3);
+  PRECONDITION(char_set[1] == '-');
```

Comment:
💡  `std::regex_match` with capture groups could be a bit easier to get my head around.
⛏ You could separate concerns - get character bounds and parse character bounds
❓ Why restrictions and not bounds?